library/alloc/src/boxed.rs

```diff
@@ -1056,8 +1056,25 @@ impl<T: ?Sized> Box<T> {
     #[stable(feature = "box_raw", since = "1.4.0")]
     #[inline]
     #[must_use = "call `drop(Box::from_raw(ptr))` if you intend to drop the `Box`"]
-    pub unsafe fn from_raw(raw: *mut T) -> Self {
-        unsafe { Self::from_raw_in(raw, Global) }
+    pub unsafe fn from_raw(ptr: *mut T) -> Self {
+        core::assert_unsafe_precondition!(
+            check_language_ub,
+            "Box::from_raw requires that its pointer argument is properly aligned and not null",
+            (
+                ptr: *const () = ptr as *const (),
+                align: usize = align_of::<T>(),
+            ) => const_eval_select!(
+                @capture { ptr: *const (), align: usize } -> bool:
+                if const {
+                    !ptr.is_null()
+                } else {
+                    ptr.is_aligned_to(align) && !ptr.is_null()
+                }
+            )
```

Comment:
We have maybe_is_aligned_and_not_null as a helper for this, why not use that?

Reply:
I can't use that, because `impl<T: ?Sized> Box<T>` means that I cannot use T::IS_ZST, which means I cannot call `maybe_is_aligned_and_not_null`.